checker/src/main/java/org/checkerframework/checker/determinism/DeterminismVisitor.java

```diff
@@ -333,6 +335,34 @@ protected void commonAssignmentCheck(
         super.commonAssignmentCheck(varType, valueType, valueTree, errorKey);
     }
 
+    /**
+     * Issues an error if a {@code Det} field is declared in a class ({@param classTree}) that is
+     * not annotated as {@code @Det}.
+     */
+    @Override
+    public void processClassTree(ClassTree classTree) {
+        List<? extends Tree> members = classTree.getMembers();
+
+        for (Tree mem : members) {
+            if (mem.getKind() == Tree.Kind.VARIABLE) {
+                AnnotatedTypeMirror fieldAnno = atypeFactory.getAnnotatedType(mem);
+                if (fieldAnno.hasAnnotation(atypeFactory.DET)) {
+                    Element classTreeElement = TreeUtils.elementFromTree(classTree);
+                    AnnotatedTypeMirror classAnno = atypeFactory.getAnnotatedType(classTreeElement);
+                    if (!classAnno.hasAnnotation(atypeFactory.DET)) {
```

Comment:
Instead if of checking this inside the for loop, do it once before checking any fields. If the class is annotated @Det, there’s no need to check the fields.